components/producers/semgrep/types/semgrep-issue.go

```diff
@@ -32,5 +40,76 @@ type SemgrepResults struct {
 // Metavars currently is empty but could represent more metavariables for semgrep.
 type Metavars struct{}
 
-// Metadata currently is empty, however, could represent semgrep issue metadata going forward.
-type Metadata struct{}
+// Metadata contains semgrep issue metadata
+type Metadata struct {
+	CWE []int32
+}
+
+func (t *Metadata) UnmarshalJSON(data []byte) error {
+	if string(data) == "null" || string(data) == `""` {
+		return nil
+	}
+
+	var realMetadata struct {
```

Comment:
I don't understand why this is so complex. As far as I see you are checking to see if you have one `CWE-[0-9]+` or multiple elements in a list. You could just check if the data you get converted into a string starts with a `CWE` otherwise try to unmarshal the bytes into a slice of strings and then convert them into `int32`s

Also there is no test for the correct JSON unmarshaling of the Metadata object